src/rgw/rgw_admin.cc

```diff
@@ -2151,10 +2155,14 @@ int main(int argc, char **argv)
         return EINVAL;
       }
     } else if (ceph_argparse_witharg(args, i, &val, "--max-size", (char*)NULL)) {
-      max_size = (int64_t)strict_strtoll(val.c_str(), 10, &err);
+      max_size = (int64_t)strict_sistrtoll(val.c_str(), &err);
       if (!err.empty()) {
-        cerr << "ERROR: failed to parse max size: " << err << std::endl;
-        return EINVAL;
+        if (!strcmp(err.c_str(), "strict_sistrtoll: value should not be negative")) {
+          max_size = -1;
+        } else {
+          cerr << "ERROR: failed to parse max size: " << err << std::endl;
+          return EINVAL;
+        }
```

Comment:
since `strict_sistrtoll()` is just a wrapper for `strict_si_cast<uint64_t>()`, i think we can just use `strict_si_cast<int64_t>()` here instead. this gives us the correct type for `max_size`, allows negative values, and doesn't require parsing the error string. can you give that a try?